### PR TITLE
Draft PR for Add fetching price from endpoint #72

### PR DESCRIPTION
**Pull request by Benjtalkshow**
Draft PR for Add fetching price from endpoint #72

### CODE REVIEW

File: web_app/contract_tools/mixins/dashboard.py [hunk 1]
```diff
@@ -9,12 +9,45 @@
...
+# [ADDED] New constant for AVNU price endpoint
+AVNU_PRICE_URL = "https://starknet.impulse.avnu.fi/v1/tokens/short"
+
 
 class DashboardMixin:
     """
     Mixin class for dashboard related methods.
     """
 
+    # [ADDED] New method to fetch current token prices
```
Comment (djeck1432) [inline web_app/contract_tools/mixins/dashboard.py:21]: Remove this comment please

File: web_app/contract_tools/mixins/dashboard.py [hunk 2]
```diff
@@ -9,12 +9,45 @@
...
+                address = token_data.get("address")
+                current_price = token_data.get("currentPrice")
+                
+                if address and current_price is not None:
+                    symbol = TokenParams.get_token_symbol(address)
+                    if symbol:
+                        prices[symbol] = float(current_price)
+
+            return prices
+        except Exception as e:
```
Comment (djeck1432) [inline web_app/contract_tools/mixins/dashboard.py:47]: Please, be more specific with Exception which you are trying to catch

**Review by djeck1432 — CHANGES_REQUESTED** [review]

**Comment by djeck1432** [comment]
@Benjtalkshow Hi, what is the update for this task, you have a little time to finish this task, otherwise it will be unassigned

**Comment by Benjtalkshow** [comment]
@Benjtalkshow 
Alright ser. 
I will make the requested changes soon. thanks

**Comment by djeck1432** [comment]
thank you, don't forget to run `pylint` command 
`pylint $(git ls-files '*.py') --disable=all --enable=C0114,C0115,C0116,C0301`
and sync with main

**Comment by djeck1432** [comment]
@Benjtalkshow this task is critical and blocking us to proceed, in 4-6 hours this task will be unassigned from you. Please take it into account

**Comment by Benjtalkshow** [comment]
> @Benjtalkshow this task is critical and blocking us to proceed, in 4-6 hours this task will be unassigned from you. Please take it into account

You will see it soon @djeck1432

**Comment by Benjtalkshow** [comment]
Hi @djeck1432 
Why am i getting this: `# pylint $(git ls-files '*.py') --disable=all --enable=C0114,C0115,C0116,C0301`
`/bin/sh: 1: git: not found`
`/bin/sh: 1: pylint: not found`

**Comment by Benjtalkshow** [comment]
Hello @djeck1432 
You can review the code now, i am active now

File: web_app/contract_tools/mixins/dashboard.py [hunk 3]
```diff
@@ -14,12 +14,44 @@
...
+
+        for token_data in response:
+            try:
+                address = token_data.get("address")
+                current_price = token_data.get("currentPrice")
+                if address and current_price is not None:
+                    symbol = TokenParams.get_token_symbol(address)
+                    if symbol:
+                        prices[symbol] = float(current_price)
+            except Exception as e:  # handle if token price parsing fails
```
Comment (djeck1432) [inline web_app/contract_tools/mixins/dashboard.py:47]: Please, be more specific with Exception, what are you going to catch here?

File: web_app/contract_tools/mixins/dashboard.py [hunk 4]
```diff
@@ -14,12 +14,44 @@
...
+            return prices
+
+        for token_data in response:
+            try:
+                address = token_data.get("address")
+                current_price = token_data.get("currentPrice")
+                if address and current_price is not None:
+                    symbol = TokenParams.get_token_symbol(address)
+                    if symbol:
+                        prices[symbol] = float(current_price)
```
Comment (djeck1432) [inline web_app/contract_tools/mixins/dashboard.py:46]: Don't use float, it's not precise enough, use Decimal or string

File: web_app/contract_tools/mixins/dashboard.py [hunk 5]
```diff
@@ -14,12 +14,44 @@
...
+AVNU_PRICE_URL = "https://starknet.impulse.avnu.fi/v1/tokens/short"
+
 
 class DashboardMixin:
     """
     Mixin class for dashboard related methods.
     """
 
+@classmethod
+async def get_current_prices(cls) -> Dict[str, float]:
```
Comment (djeck1432) [inline web_app/contract_tools/mixins/dashboard.py:27]: where is a 4  spaces? you have an error

**Review by djeck1432 — CHANGES_REQUESTED** [review]

File: web_app/contract_tools/mixins/dashboard.py (hunk 5 above)
Comment (Benjtalkshow) [reply web_app/contract_tools/mixins/dashboard.py:27]: Alright ser

**Comment by djeck1432** [comment]
@Benjtalkshow Any updates, again this task is blocking other developer, please finalize it as soon as possible

**Comment by Benjtalkshow** [comment]
> @Benjtalkshow Any updates, again this task is blocking other developer, please finalize it as soon as possible

Hello @djeck1432 
I updated the code already. Check Please

**Review by djeck1432 — APPROVED** [review]
Looks good to me

**Pull request merged**